### PR TITLE
Release 5.23.0

### PR DESCRIPTION
**Pull request by dktapps**
This includes all changes from minor-next as of December 5th

## !!!! DO NOT squash merge this PR !!!!

### CODE REVIEW

File: changelogs/5.23.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+**WARNING: If your plugin uses the `pocketmine\network\mcpe` namespace, you're not shielded by API change constraints.**
+Consider using the `mcpe-protocol` directive in `plugin.yml` as a constraint if you're using packets directly.
+
+## General
+- `/timings` now supports collecting timings from async task workers. These new timings will be shown alongside `Full Server Tick` timings, but will not be counted in total load.
+- Added `/xp` command.
+- `start.sh` will now emit warnings when the server process exits with an unusual exit code. This helps to detect unexpected segmentation faults and other kinds of native errors.
+
+## Gameplay
+- Added the following new items:
```
Comment (ipad54) [inline changelogs/5.23.md:19]: You haven’t included the ice bomb in this list

Comment (dktapps) [reply changelogs/5.23.md:19]: Good catch

File: changelogs/5.23.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  - Timings cannot directly control timings on other threads, so these callbacks allow plugins to use custom mechanisms to toggle, reset and collect timings.
+  - PocketMine-MP currently uses this to collect timings from async task workers. More internal threads may be supported in the future.
+- The following API methods have been added:
+  - `public static TimingsHandler::getCollectCallbacks() : ObjectSet<\Closure() : list<Promise<list<string>>>` - callbacks for (asynchronously) collecting timings (typically from other threads). The returned promises should be resolved with the result of `TimingsHandler::printCurrentThreadRecords()`.
+  - `public static TimingsHandler::getReloadCallbacks() : ObjectSet<\Closure() : void>` - callbacks called when timings are reset
+  - `public static TimingsHandler::getToggleCallbacks() : ObjectSet<\Closure(bool $enable) : void>` - callbacks called when timings are enabled/disabled
+  - `public static TimingsHandler::requestPrintTimings() : Promise<list<string>>` - asynchronously collects timing results from all threads and assembles them into a single report
+- The following API methods have been deprecated:
```
Comment (ipad54) [inline changelogs/5.23.md:88]: I’ve also observed now that there is no mention about deprecation `Timings::getAsyncTaskErrorTimings()` and `Timings::GROUP_BREAKDOWN`

Comment (dktapps) [reply changelogs/5.23.md:88]: Those are internal, so I didn't think it was necessary to mention them. Really, the problem is that they've never been marked as such.

**Review by ipad54 — APPROVED** [review]

**Pull request merged**